m3u8/reader.go

```diff
@@ -163,6 +163,52 @@ func (p *MediaPlaylist) SetVersion(ver uint8) {
 	p.ver = ver
 }
 
+func (p *MediaPlaylist) LastSegIndex() uint64 {
+	nextSeqNo := p.NextMSNIndex + p.AlreadySkippedSegs
+	if p.NextPartIndex == 0 {
+		// Just rolled over to the next segment
+		return nextSeqNo - 1
+	}
+	return nextSeqNo
+}
+
+func (p *MediaPlaylist) LastPartSegIndex() uint64 {
+	if p.NextPartIndex == 0 {
+		// Just rolled over to the next segment
+		return p.MaxPartIndex - 1
```

Comment:
This looks inconsistent to me. The last index before 0, should be `MaxPartIndex` not `MaxPartIndex - 1`